ykrt/src/deopt.rs

```diff
@@ -302,7 +375,30 @@ unsafe extern "C" fn __ykrt_deopt(
     // pass in variables from this scope via a struct which is passed into the function.
     LLVMOrcThreadSafeModuleWithModuleDo(moduleref, ts_reconstruct, infoptr as *mut c_void);
 
+    // We want to start side tracing only after we deoptimised so we don't trace the whole deopt
+    // routine which will later be costly to disassemble.
+    if guardid != SIDETRACE_LAST_GUARD_ID {
+        let guard = &ctr.guards[guardid];
+        if guard.failcount() >= ctr.mt.sidetrace_threshold() - 1 {
```

Comment:
What happens if `sidetrace_threshold` is 0?

Reply:
Whoops. Will change it to `>`.

Reply:
Fixed in c0a151a9e20e9